barman/infofile.py

```diff
@@ -792,3 +794,25 @@ def save(self, filename=None, file_object=None):
             if not os.path.exists(dir_name):
                 os.makedirs(dir_name)
         super(LocalBackupInfo, self).save(filename=filename, file_object=file_object)
+
+    def get_backup_manifest_path(self):
+        return os.path.join(self.get_data_directory(), "backup_manifest")
+
+    def get_parent_backup_info(self):
+        if self.parent_backup_id:
+            return LocalBackupInfo(
+                self.server,
+                backup_id=self.parent_backup_id,
+            )
+
+        return None
+
+    def get_child_backup_info(self, child_backup_id):
+        if self.children_backup_ids:
+            if child_backup_id in self.children_backup_ids:
+                return LocalBackupInfo(
+                    self.server,
+                    backup_id=child_backup_id,
+                )
+
+        return None
```

Comment:
I think we need to check the for "empty" backup info.

The implementation seems to be working well, but if the parent or child backup doens't exist anymore, we will get an empty local backup info, e.g.:

```python
LocalBackupInfo(backup_label=None, begin_offset=None, begin_time=None, begin_wal=None, begin_xlog=None, children_backup_ids=None, compression=None, config_file=None, copy_stats=None, deduplicated_size=None, end_offset=None, end_time=None, end_wal=None, end_xlog=None, error=None, hba_file=None, ident_file=None, included_files=None, mode='postgres', parent_backup_id=None, pgdata=None, server_name='pg17-postgres-default', size=None, status='EMPTY', summarize_wal=None, systemid=None, tablespaces=None, timeline=None, version=None, xlog_segment_size=16777216)
```

I'd say to also return `None` here if the `status` is `EMPTY`.

What do you think?